setup.py

```diff
@@ -1,41 +1,137 @@
 #!/usr/bin/env python
 from __future__ import print_function
 import numpy
+import os, sys
 
 # import version from file
 with open('pyroomacoustics/version.py') as f:
     exec(f.read())
 
 try:
-    from setuptools import setup
-    from setuptools import Extension
+    from setuptools import setup, Extension
+    from setuptools.command.build_ext import build_ext
+    from setuptools import distutils
 except ImportError:
     print("Setuptools unavailable. Falling back to distutils.")
+    import distutils
     from distutils.core import setup
     from distutils.extension import Extension
+    from distutils.command.build_ext import build_ext
 
 from Cython.Build import cythonize
 
 # To use a consistent encoding
 from codecs import open
 from os import path
 
-# build C extension for image source model
-src_dir = 'pyroomacoustics/c_package'
-files = ['libroom.c', 'wall.c', 'linalg.c', 'room.c', 'is_list.c', 'shoebox.c']
 
-libroom_ext = Extension('pyroomacoustics.c_package.libroom',
-                    extra_compile_args = ['-Wall', '-O3', '-std=c99'],
-                    sources = [src_dir + '/' + f for f in files],
-                    include_dirs=[src_dir,numpy.get_include()])
-cython_ext = Extension("pyroomacoustics.build_rir", ["pyroomacoustics/build_rir.pyx"])
+class get_pybind_include(object):
+    """Helper class to determine the pybind11 include path
+
+    The purpose of this class is to postpone importing pybind11
+    until it is actually installed, so that the ``get_include()``
+    method can be invoked. """
+
+    def __init__(self, user=False):
+        self.user = user
+
+    def __str__(self):
+        import pybind11
+        return pybind11.get_include(self.user)
+
+
+# build C extension for image source model
+libroom_src_dir = 'pyroomacoustics/libroom_src'
+ext_modules = [
+        Extension(
+            'pyroomacoustics.libroom',
+            [ os.path.join(libroom_src_dir, f)
+                for f in ['libroom.cpp','wall.cpp', 'geometry.cpp', 'room.cpp']],
+            include_dirs=[
+                '.',
+                libroom_src_dir,
+                str(get_pybind_include()),
+                str(get_pybind_include(user=True)),
+                os.path.join(libroom_src_dir, 'ext/eigen'),
```

Comment:
Would be good to put a note somewhere that the following needs to be run so that the `Eigen` library files are in the proper place:

`git submodule update --init --recursive`

More on this [here](https://stackoverflow.com/questions/10168449/git-update-submodules-recursively).

Reply:
Definitely! I will add this to the README and doc.